### PR TITLE
Fix small bugs for splits.

### PR DESCRIPTION
**Pull request by ccl-core**
* Add `wd:` to the context when needed
* Fix small bug in regex transformation when dealing with paths

### CODE REVIEW

File: datasets/1.0/audio_test/metadata.json [hunk 1]
```diff
@@ -42,7 +42,8 @@
     "separator": "cr:separator",
     "source": "cr:source",
     "subField": "cr:subField",
-    "transform": "cr:transform"
+    "transform": "cr:transform",
+    "wd": "https://www.wikidata.org/wiki/"
```
Comment (marcenacp) [inline datasets/1.0/audio_test/metadata.json:46]: wd shouldn't be in the default context. I'm of the opinion that it should be added only when needed (the context is long enough). We can discuss this.

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py [hunk 2]
```diff
@@ -34,6 +35,8 @@ def _apply_transform_fn(value: Any, transform: Transform, field: Field) -> Any:
     """Applies one transform to `value`."""
     if transform.regex is not None:
         source_regex = re.compile(transform.regex)
+        if isinstance(value, pathlib.PurePath):
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py:38]: Always do:

```python
value = os.fspath(value)
```

?

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/field_test.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 """field_test module."""
 
+import pathlib
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/operation_graph/operations/field_test.py:3]: Use epath.Path everywhere?

File: datasets/1.0/huggingface-anthropic-hh-rlhf/metadata.json [hunk 4]
```diff
@@ -266,6 +303,31 @@
       "@id": "harmless-base",
       "name": "harmless-base",
       "field": [
+        {
+          "@type": "cr:Field",
+          "@id": "harmless-base/split",
```
Comment (marcenacp) [inline datasets/1.0/huggingface-anthropic-hh-rlhf/metadata.json:308]: Are the fields repeated between the RecordSets?

**Review by marcenacp — CHANGES_REQUESTED** [review]
Great, thanks!

File: datasets/1.0/huggingface-anthropic-hh-rlhf/metadata.json (hunk 4 above)
Comment (ccl-core) [reply datasets/1.0/huggingface-anthropic-hh-rlhf/metadata.json:308]: Exactly -- is there a better way to express this with less redundancy?

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py (hunk 2 above)
Comment (ccl-core) [reply python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py:38]: Done.

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/field_test.py (hunk 3 above)
Comment (ccl-core) [reply python/mlcroissant/mlcroissant/_src/operation_graph/operations/field_test.py:3]: In tests sure. In `field.py` not sure, I'd need to convert `fullpath` to be epath.Path as it is a `pathlib.PurePath` (https://github.com/mlcommons/croissant/blob/0f95e04763557929e4f4c6711c108c0d9cf7b818/python/mlcroissant/mlcroissant/_src/core/path.py#L24)

Comment (marcenacp) [reply python/mlcroissant/mlcroissant/_src/operation_graph/operations/field_test.py]: Agreed.

File: datasets/1.0/huggingface-anthropic-hh-rlhf/metadata.json (hunk 4 above)
Comment (marcenacp) [reply datasets/1.0/huggingface-anthropic-hh-rlhf/metadata.json:308]: We could parametrize the config name. But that'd be a feature in itself, so no need to do this in this PR. Good to know we have a dataset to develop this feature!

**Review by marcenacp — APPROVED** [review]
Thanks!

File: datasets/1.0/audio_test/metadata.json (hunk 1 above)
Comment (ccl-core) [reply datasets/1.0/audio_test/metadata.json]: Sure. Done.

**Pull request merged**